Rework extensions guides (#1802)
kontena/lens: docs/extensions/guides/working-with-mobx.md

# Working with MobX

## Introduction

> Lens uses MobX on top of React's state management system. The result is a more declarative state management style, rather than React's native `setState` mechanism. You can review how React handles state management [here](https://reactjs.org/docs/faq-state.html). The following is a quick overview:

Reviewer: To help with reviews, would you mind splitting this? Namely, having one sentence on each line

> * `React.Component` is generic with respect to both `props` and `state` (with a default of empty object types).

Reviewer: ```suggestion
* `React.Component` is generic with respect to both `props` and `state` (which default to the empty object type).
```

* `props` should be considered read-only from the point of view of the component, and it is the mechanism for passing in arguments to a component.

> * `state` is a component's internal state, and can be read by accessing the parent field `state`.

Reviewer: ```suggestion
* `state` is a component's internal state, and can be read by accessing the super-class field `state`.
```

there is nothing "parent" about
